### PR TITLE
Ensure user_properties is a list

### LINKED ISSUE #4034: TestReport.user_properties is documented as a list, actually a tuple
https://github.com/pytest-dev/pytest/blob/253419316ce09b952862811db229a9e9383a332b/src/_pytest/reports.py#L137-L139

This makes me nervous when using it in plugins, e.g. to pass Hypothesis statistics around in HypothesisWorks/hypothesis#1577.  My code handles either a list or a tuple here, but it would be nice to - one way or another - make the code and the documentation consistent.

### PR DESCRIPTION
**Pull request by Zac-HD**
Closes #4034.

### CODE REVIEW

File: src/_pytest/reports.py [hunk 1]
```diff
@@ -136,6 +136,11 @@ def __init__(
 
         #: user properties is a list of tuples (name, value) that holds user
         #: defined properties of the test
+        if user_properties is None:
```
Comment (nicoddemus) [inline src/_pytest/reports.py]: I think we can make this more succinct:

```python
user_properties = [] if user_properties is None else list(user_properties)

**Comment by Zac-HD** [comment]
Happy to help!  I use pytest all the time, so it's nice to give some code back too :smile:

**Comment by Zac-HD** [comment]
@nicoddemus - are these coverage tools ever useful?  Because the diff clearly doesn't change coverage at all, but one tool has it going up and the other going down!

**Comment by nicoddemus** [comment]
> are these coverage tools ever useful? 

We are still evaluating them... can't really answer that at the moment. 🤔

**Comment by Zac-HD** [comment]
Fair enough!  My thoughts:

- Coverage tools that give inconsistent results are just as bad as any other flaky test.
- The important thing about coverage measurement is not percentage, it's "which lines are not covered?"
- 100% is therefore the only percentage target that makes sense; perhaps for a subset of files.

**Comment by Zac-HD** [comment]
@nicoddemus - Appveyor failure looks like a job failed to upload coverage data, not failing tests :relieved:

**Comment by nicoddemus** [comment]
> Fair enough! My thoughts:

Definitely, agree with all your points. If `codecov` keeps behaving like that, I'm afraid we will have to revert back to `coveralls` (this was discussed briefly in https://github.com/tox-dev/tox/pull/1019#issuecomment-424837884 also).

**Pull request merged**